src/experimentClient.ts

```diff
@@ -136,10 +137,18 @@ export class ExperimentClient implements Client {
       storage,
     );
     this.flags = getFlagStorage(this.apiKey, this.config.instanceName, storage);
-    // eslint-disable-next-line no-void
-    void this.flags.load();
-    // eslint-disable-next-line no-void
-    void this.variants.load();
+    this.flagsAndVariantsLoadedPromise = [
+      this.flags.load(this.getInitialFlags()),
+      this.variants.load(),
+    ];
+  }
+
+  /**
+   * Call to ensure the completion of the loading variants and flags from localStorage upon initialization.
+   */
+  public async ready(): Promise<ExperimentClient> {
```

Comment:
can we rename this to `cacheReady()` I think then it's more clear the difference between this and `start()` promise resolving.